### PR TITLE
Potential Fix for failing frame doctest

### LINKED ISSUE #20955: Doctest failure in sympy/physics/vector/frame.py
See discussion in #20946.

CC @moorepants 

There was a doctest failure but the test was disabled in #20954 to unblock CI. The cause of the failure remains unfixed though.

The failure was:
```
$ bin/doctest sympy/physics/vector/
====================================================== test process starts =======================================================
executable:         /Users/enojb/current/sympy/sympy/venv/bin/python  (3.8.5-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       gmpy 2.0.8
numpy:              None
hash randomization: on (PYTHONHASHSEED=3249984020)

sympy/physics/vector/functions.py[9] .........                                                                                [OK]
sympy/physics/vector/vector.py[14] ..............                                                                             [OK]
sympy/physics/vector/point.py[13] .............                                                                               [OK]
sympy/physics/vector/frame.py[15] .....F.........                                                                           [FAIL]
sympy/physics/vector/fieldfunctions.py[7] .......                                                                             [OK]
sympy/physics/vector/dyadic.py[10] ..........                                                                                 [OK]
sympy/physics/vector/printing.py[4] ....                                                                                      [OK]

__________________________________________________________________________________________________________________________________
__________________________________ sympy.physics.vector.frame.ReferenceFrame.orient_space_fixed __________________________________
File "/Users/enojb/current/sympy/sympy/sympy/physics/vector/frame.py", line 838, in sympy.physics.vector.frame.ReferenceFrame.orient_space_fixed
Failed example:
    B.dcm(N).simplify()
Expected:
    Matrix([
    [ sin(q1)*sin(q2)*sin(q3) + cos(q1)*cos(q3), sin(q1)*cos(q2), sin(q1)*sin(q2)*cos(q3) - sin(q3)*cos(q1)],
    [-sin(q1)*cos(q3) + sin(q2)*sin(q3)*cos(q1), cos(q1)*cos(q2), sin(q1)*sin(q3) + sin(q2)*cos(q1)*cos(q3)],
    [                           sin(q3)*cos(q2),        -sin(q2),                           cos(q2)*cos(q3)]])
Got:
    Matrix([
    [ sin(q1)*sin(q2)*sin(q3) + cos(q1)*cos(q3), sin(q1)*cos(q2),                                                                                sin(q1)*sin(q2)*cos(q3) - sin(q3)*cos(q1)],
    [-sin(q1)*cos(q3) + sin(q2)*sin(q3)*cos(q1), cos(q1)*cos(q2), sin(-q1 + q2 + q3)/4 - sin(q1 - q2 + q3)/4 + sin(q1 + q2 - q3)/4 + sin(q1 + q2 + q3)/4 + cos(q1 - q3)/2 - cos(q1 + q3)/2],
    [                           sin(q3)*cos(q2),        -sin(q2),                                                                                                          cos(q2)*cos(q3)]])
```

### PR DESCRIPTION
**Pull request by sidhu1012**
#### References to other Issues or PRs

Fixe #20955

#### Brief description of what is fixed or changed
Added a newframe `A` to doctest.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/physics/vector/frame.py [hunk 1]
```diff
@@ -834,8 +835,8 @@ def orient_space_fixed(self, parent, angles, rotation_order):
 
         >>> B1.orient_axis(N, N.z, q1)
         >>> B2.orient_axis(B1, N.x, q2)
-        >>> B.orient_axis(B2, N.y, q3)
-        >>> B.dcm(N).simplify() # doctest: +SKIP
+        >>> A.orient_axis(B2, N.y, q3)
+        >>> A.dcm(N).simplify() # doctest: +SKIP
```
Comment (0sidharth) [inline sympy/physics/vector/frame.py:839]: Shouldn't the SKIP tag be removed before checking?

Comment (sidhu1012) [reply sympy/physics/vector/frame.py:839]: Oh yes, my bad.

**Comment by moorepants** [comment]
I've tried this locally in a few ways and it didn't fix the problem. Does it pass locally for you?

File: sympy/physics/vector/frame.py (hunk 1 above)
Comment (moorepants) [inline sympy/physics/vector/frame.py:839]:
```suggestion
        >>> A.dcm(N).simplify()
```

**Comment by oscarbenjamin** [comment]
I don't think we can say that this "fixes" the issue by changing the doctest. There's clearly a deeper cacheing problem here that needs to be investigated.

**Comment by moorepants** [comment]
Yes, we need to leave the issue open even if this happens to work.

**Comment by moorepants** [comment]
Here was the change I made locally that was similar to this but still causes a failure:

```
diff --git a/sympy/physics/vector/frame.py b/sympy/physics/vector/frame.py
index d3866df2e8..9a6ff538e4 100644
--- a/sympy/physics/vector/frame.py
+++ b/sympy/physics/vector/frame.py
@@ -712,6 +712,7 @@ def orient_body_fixed(self, parent, angles, rotation_order):
         >>> B = ReferenceFrame('B')
         >>> B1 = ReferenceFrame('B1')
         >>> B2 = ReferenceFrame('B2')
+        >>> B3 = ReferenceFrame('B3')
 
         For example, a classic Euler Angle rotation can be done by:
 
@@ -729,8 +730,8 @@ def orient_body_fixed(self, parent, angles, rotation_order):
 
         >>> B1.orient_axis(N, N.x, q1)
         >>> B2.orient_axis(B1, B1.y, q2)
-        >>> B.orient_axis(B2, B2.x, q3)
-        >>> B.dcm(N)
+        >>> B3.orient_axis(B2, B2.x, q3)
+        >>> B3.dcm(N)
         Matrix([
         [        cos(q2),                            sin(q1)*sin(q2),                           -sin(q2)*cos(q1)],
         [sin(q2)*sin(q3), -sin(q1)*sin(q3)*cos(q2) + cos(q1)*cos(q3),  sin(q1)*cos(q3) + sin(q3)*cos(q1)*cos(q2)],
@@ -819,9 +820,11 @@ def orient_space_fixed(self, parent, angles, rotation_order):
         >>> from sympy.physics.vector import ReferenceFrame
         >>> q1, q2, q3 = symbols('q1, q2, q3')
         >>> N = ReferenceFrame('N')
+        >>> N2 = ReferenceFrame('N2')
         >>> B = ReferenceFrame('B')
         >>> B1 = ReferenceFrame('B1')
         >>> B2 = ReferenceFrame('B2')
+        >>> B3 = ReferenceFrame('B3')
 
         >>> B.orient_space_fixed(N, (q1, q2, q3), '312')
         >>> B.dcm(N)
@@ -832,10 +835,10 @@ def orient_space_fixed(self, parent, angles, rotation_order):
 
         is equivalent to:
 
-        >>> B1.orient_axis(N, N.z, q1)
-        >>> B2.orient_axis(B1, N.x, q2)
-        >>> B.orient_axis(B2, N.y, q3)
-        >>> B.dcm(N).simplify()
+        >>> B1.orient_axis(N2, N2.z, q1)
+        >>> B2.orient_axis(B1, N2.x, q2)
+        >>> B3.orient_axis(B2, N2.y, q3)
+        >>> B3.dcm(N2).simplify()
         Matrix([
         [ sin(q1)*sin(q2)*sin(q3) + cos(q1)*cos(q3), sin(q1)*cos(q2), sin(q1)*sin(q2)*cos(q3) - sin(q3)*cos(q1)],
         [-sin(q1)*cos(q3) + sin(q2)*sin(q3)*cos(q1), cos(q1)*cos(q2), sin(q1)*sin(q3) + sin(q2)*cos(q1)*cos(q3)],
@@ -845,15 +848,18 @@ def orient_space_fixed(self, parent, angles, rotation_order):
         related by the order of the rotations, i.e. the reverse order of body
         fixed will give space fixed and vice versa.
 
-        >>> B.orient_space_fixed(N, (q1, q2, q3), '231')
-        >>> B.dcm(N)
+        >>> N3 = ReferenceFrame('N3')
+        >>> A1 = ReferenceFrame('A1')
+        >>> A2 = ReferenceFrame('A2')
+        >>> A1.orient_space_fixed(N3, (q1, q2, q3), '231')
+        >>> A1.dcm(N3)
         Matrix([
         [cos(q1)*cos(q2), sin(q1)*sin(q3) + sin(q2)*cos(q1)*cos(q3), -sin(q1)*cos(q3) + sin(q2)*sin(q3)*cos(q1)],
         [       -sin(q2),                           cos(q2)*cos(q3),                            sin(q3)*cos(q2)],
         [sin(q1)*cos(q2), sin(q1)*sin(q2)*cos(q3) - sin(q3)*cos(q1),  sin(q1)*sin(q2)*sin(q3) + cos(q1)*cos(q3)]])
 
-        >>> B.orient_body_fixed(N, (q3, q2, q1), '132')
-        >>> B.dcm(N)
+        >>> A2.orient_body_fixed(N3, (q3, q2, q1), '132')
+        >>> A2.dcm(N3)
         Matrix([
         [cos(q1)*cos(q2), sin(q1)*sin(q3) + sin(q2)*cos(q1)*cos(q3), -sin(q1)*cos(q3) + sin(q2)*sin(q3)*cos(q1)],
         [       -sin(q2),                           cos(q2)*cos(q3),                            sin(q3)*cos(q2)],
moorepants@nandi:sympy(orient-doctest)$ bin/doctest sympy/physics/vector/
================================================================================================================================== test process starts ===================================================================================================================================
executable:         /home/moorepants/miniconda/bin/python  (3.7.9-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       gmpy 2.1.0b1
numpy:              1.20.1
hash randomization: on (PYTHONHASHSEED=2148361169)

sympy/physics/vector/functions.py[9] .........                                                                                                                                                                                                                                        [OK]
sympy/physics/vector/dyadic.py[10] ..........                                                                                                                                                                                                                                         [OK]
sympy/physics/vector/frame.py[15] .....F.........                                                                                                                                                                                                                                   [FAIL]
sympy/physics/vector/fieldfunctions.py[7] .......                                                                                                                                                                                                                                     [OK]
sympy/physics/vector/vector.py[14] ..............                                                                                                                                                                                                                                     [OK]
sympy/physics/vector/point.py[13] .............                                                                                                                                                                                                                                       [OK]
sympy/physics/vector/printing.py[4] ....                                                                                                                                                                                                                                              [OK]

__________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________
______________________________________________________________________________________________________________ sympy.physics.vector.frame.ReferenceFrame.orient_space_fixed ______________________________________________________________________________________________________________
File "/home/moorepants/src/sympy/sympy/physics/vector/frame.py", line 841, in sympy.physics.vector.frame.ReferenceFrame.orient_space_fixed
Failed example:
    B3.dcm(N2).simplify()
Expected:
    Matrix([
    [ sin(q1)*sin(q2)*sin(q3) + cos(q1)*cos(q3), sin(q1)*cos(q2), sin(q1)*sin(q2)*cos(q3) - sin(q3)*cos(q1)],
    [-sin(q1)*cos(q3) + sin(q2)*sin(q3)*cos(q1), cos(q1)*cos(q2), sin(q1)*sin(q3) + sin(q2)*cos(q1)*cos(q3)],
    [                           sin(q3)*cos(q2),        -sin(q2),                           cos(q2)*cos(q3)]])
Got:
    Matrix([
    [ sin(q1)*sin(q2)*sin(q3) + cos(q1)*cos(q3), sin(q1)*cos(q2),                                                                                sin(q1)*sin(q2)*cos(q3) - sin(q3)*cos(q1)],
    [-sin(q1)*cos(q3) + sin(q2)*sin(q3)*cos(q1), cos(q1)*cos(q2), sin(-q1 + q2 + q3)/4 - sin(q1 - q2 + q3)/4 + sin(q1 + q2 - q3)/4 + sin(q1 + q2 + q3)/4 + cos(q1 - q3)/2 - cos(q1 + q3)/2],
    [                           sin(q3)*cos(q2),        -sin(q2),                                                                                                          cos(q2)*cos(q3)]])

================================================================================================================== tests finished: 71 passed, 1 failed, in 9.19 seconds ==================================================================================================================
DO *NOT* COMMIT!
```

**Pull request closed without merging**